praxis-isncsci/ui
# Select cell with mouse

Closes #110 

## Problem

The most basic way to select cells on the ISNCSCI input grid is by clicking with the mouse. The behaviour needs to be implemented.

## Solution

I don't want the grid component to have to keep track of selection. We have an application state that should track that, that way it can be shared amongst components, like grids and diagram. The input controller will capture the clicks and call the appropriate use case to update the selection. The views will be updated when the states change and will receive the names associated to the cells selected through an attribute (events up, attributes down).

## Test

- [x] 1. Can select a cell by clicking on it

<details>
  <summary>Test case</summary>

  1. `npm run build`
  2. `npm start`
  3. Navigate to `http://localhost:44333/`
  4. On the grid, click on a cell
  5. Confirm that it is highlighted with a purple border
  6. Click on a different cell
  7. Confirm the previous selection gets cleared and the new cell gets highlighted

</details>

---

- [x] Multiple cells can be highlighted on a grid

<details>
  <summary>Test case</summary>

  1. Navigate to the **Storybook** story `http://localhost:6006/?path=/story/webcomponents-praxisisncscigrid--primary`
  2. In the controls panel, click on one of the options
  3. Confirm the cell on the grid gets highlighted
  4. Press shift and select all the options
  5. Confirm that the selected sells appear highlighted

  <img alt="example of multiple cells selected" width="500" src="https://github.com/praxis-isncsci/ui/assets/1294355/dfcc01f6-8abf-4edd-b768-55ba76f1fa45" />

</details>
